dandi/dandiapi.py

```diff
@@ -514,11 +520,20 @@ def check_schema_version(self, schema_version: Optional[str] = None) -> None:
 
     def get_asset(self, asset_id: str) -> "BaseRemoteAsset":
         """
-        Fetch the asset with the given asset ID.  The returned object will not
-        have any information about the Dandiset associated with the asset; for
-        that, the `RemoteDandiset.get_asset()` method must be used instead.
+        Fetch the asset with the given asset ID.  If the given asset does not
+        exist, a `NotFoundError` is raised.
+
+        The returned object will not have any information about the Dandiset
+        associated with the asset; for that, the `RemoteDandiset.get_asset()`
+        method must be used instead.
         """
-        return BaseRemoteAsset._from_metadata(self, self.get(f"/assets/{asset_id}"))
+        try:
+            return BaseRemoteAsset._from_metadata(self, self.get(f"/assets/{asset_id}"))
+        except requests.HTTPError as e:
+            if e.response.status_code == 404:
+                raise NotFoundError(f"No such asset: {asset_id}")
```

Comment:
minor, I wonder why use `!r` for dandiset error message but not for this (asset) one?